src/AgentCommerce/SalesChannel/CreateCartRoute.php

```diff
@@ -7,24 +7,128 @@
+use Swag\PayPal\AgentCommerce\Util\PayPalCartFactory;
+use Swag\PayPal\AgentCommerce\Util\PayPalCartTransformer;
+use Swag\PayPal\AgentCommerce\Util\ShopwareCartTransformer;
 use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Attribute\Route;
 
 /**
  * @internal
  */
 #[Package('checkout')]
 #[Route(defaults: ['_routeScope' => ['paypal-agent'], '_agentScope' => [AgentSource::SCOPE_CART]])]
-class CreateCartRoute
+class CreateCartRoute extends AbstractAgentCommerceRoute
 {
+    /**
+     * @param EntityRepository<ProductCollection> $productRepository
+     */
+    public function __construct(
+        protected SalesChannelContextService $contextService,
+        private readonly EntityRepository $productRepository,
+        private readonly CartService $cartService,
+        private readonly PayPalCartTransformer $payPalCartTransformer,
+        private readonly ShopwareCartTransformer $shopwareCartTransformer,
+        private readonly RegisterRoute $registerRoute,
+        private readonly ProductLineItemFactory $lineItemFactory
+    ) {
+    }
+
     #[Route('/api/paypal/v1/merchant-cart', name: 'api.paypal.merchant-cart', methods: [Request::METHOD_POST])]
-    public function createCart(Request $request, Context $context): AgentCartResponse
+    public function createCart(Request $request, SalesChannelContext $salesChannelContext): AgentCartResponse
+    {
+        if (!$productExport = $salesChannelContext->getSalesChannel()->getProductExports()?->first()) {
+            throw AgentException::requiredFieldsMissing('product export');
+        }
+
+        $payPalCart = (new PayPalCartFactory())->create($request->getPayload()->all());
+        if ($payPalCart->getCustomer() && !$salesChannelContext->getCustomer()) {
+            $salesChannelContext = $this->loginCustomer($payPalCart, $salesChannelContext);
+        }
+
+        $mapped = [];
+        foreach ($payPalCart->getItems() as $item) {
+            if (Uuid::isValid($item->getVariantId() ?? '')) {
+                $mapped[$item->getVariantId()] = $item->getQuantity();
+            }
+        }
+
+        $criteria = new Criteria(array_keys($mapped));
+        $criteria->addFilter(
+            new EqualsFilter('streams.id', $productExport->getProductStreamId()),
+            new ProductAvailableFilter($salesChannelContext->getSalesChannelId(), ProductVisibilityDefinition::VISIBILITY_LINK)
+        );
+        $productIds = $this->productRepository->searchIds($criteria, $salesChannelContext->getContext());
+        if (!$productIds->getTotal()) {
+            throw AgentException::requiredFieldsMissing('items');
+        }
+
+        $lineItems = [];
+        /** @var string $productId */
+        foreach ($productIds->getIds() as $productId) {
+            $lineItems[] = $this->lineItemFactory->create(['id' => $productId, 'quantity' => $mapped[$productId] ?? 0], $salesChannelContext);
+        }
```

Comment:
As we now have UUIDs, we can let the ProductCartProcessor filter out invalid products:
```suggestion
        $lineItems = array_map(fn (string $productId) => $this->lineItemFactory->create(['id' => $productId, 'quantity' => $mapped[$productId] ?? 0], $salesChannelContext), array_keys($mapped));
```